xdsl/backend/riscv/register_allocation.py

```diff
@@ -116,6 +120,54 @@ def allocate(self, reg: SSAValue) -> bool:
 
         return False
 
+    def allocate_same(self, vals: Sequence[SSAValue]) -> bool:
```

Comment:
Could you leave a doc comment here explaining more in detail what it does?

My initial thought of the name was that it would make sure that it'd allocate the same register for all `vals` but it seems to be smarter than that and if one consistent register exists that has been allocated then it seemingly just uses that. 

I think this is worth documenting as this can cause incorrectness issues if misused.

Reply:
Yep good idea